src/components/Carousel.tsx

```diff
@@ -1,26 +1,117 @@
-import React from 'react';
+import React, { useRef, useState } from 'react';
 import './Carousel.scss';
+import cn from 'classnames';
 
-const Carousel: React.FC = () => (
-  <div className="Carousel">
-    <ul className="Carousel__list">
-      <li>
-        <img src="./img/1.png" alt="1" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="2" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="3" />
-      </li>
-      <li>
-        <img src="./img/1.png" alt="4" />
-      </li>
-    </ul>
-
-    <button type="button">Prev</button>
-    <button type="button">Next</button>
-  </div>
-);
+type CarouselProps = {
+  images: string[];
+  step: number;
+  itemWidth: number;
+  frameSize: number;
+  animationDuration: number;
+  infinite?: boolean;
+};
+
+const Carousel = ({
+  images,
+  step,
+  itemWidth,
+  frameSize,
+  animationDuration,
+  infinite,
+}: CarouselProps): JSX.Element => {
+  const [offset, setOffset] = useState(0);
+  const carouselListRef = useRef<HTMLUListElement>(null);
+
+  const gap = 20;
+  const maxNextTransform = -(images.length - frameSize) * (itemWidth + gap);
+
+  const handleNextClick = () => {
+    if (carouselListRef.current && offset > maxNextTransform) {
+      const newOffset = Math.max(
+        offset - step * (itemWidth + gap),
+        maxNextTransform,
+      );
+
+      setOffset(newOffset);
+    }
+
+    if (carouselListRef.current && offset === maxNextTransform && infinite) {
+      setOffset(0);
+    }
+  };
+
+  const handlePrevClick = () => {
+    if (carouselListRef.current && offset < 0) {
+      const newOffset = Math.min(offset + step * (itemWidth + gap), 0);
+
+      setOffset(newOffset);
+    }
+
+    if (carouselListRef.current && offset === 0 && infinite) {
+      setOffset(maxNextTransform);
+    }
+  };
+
+  const prevButtonClasses = cn({
+    Carousel__button: true,
+    'Carousel__button--prev': true,
+    'Carousel__button--disabled': offset === 0 && !infinite,
+  });
+
+  const nextButtonClasses = cn({
+    Carousel__button: true,
+    'Carousel__button--next': true,
+    'Carousel__button--disabled': offset === maxNextTransform && !infinite,
+  });
```

Comment:
This can be also used like that
```suggestion
    const nextButtonClasses = cn('Carousel__button--next Carousel__button', {
      'Carousel__button--disabled': offset === maxNextTransform && !infinite,
    });
```